app/src/main/java/com/kelvin/bdaypost/birthday/add/AddBirthdayFragment.kt

```diff
@@ -0,0 +1,159 @@
+import timber.log.Timber
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [AddBirthdayFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class AddBirthdayFragment : Fragment() {
+    private var _addBdayBinding: FragmentAddBirthdayBinding? = null
+    // only use this between onCreateView and onDestroyView
+    private val addBdayBinding
+        get() = _addBdayBinding!!
+
+    private var addBdayVM: AddBirthdayViewModel? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _addBdayBinding = FragmentAddBirthdayBinding.inflate(inflater, container, false)
+        return addBdayBinding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // create VM
+        addBdayVM = activity?.run {
+            ViewModelProvider(this, AddBirthdayViewModel.AddBirthdayVMFactory())
+                .get(AddBirthdayViewModel::class.java)
+        }
+        // Add VM observers
+        addBdayVM?.run {
+            this.validContactForm.observe(
+                this@AddBirthdayFragment.viewLifecycleOwner,
+                Observer { observeContactState(it) }
+            )
+
+            this.addContactResult.observe(
+                this@AddBirthdayFragment.viewLifecycleOwner,
+                Observer { observeAddContact(it) }
+            )
+        }
+
+
+        // configure date picker for birthday spinner
+        context?.let { addDatePicker(it) }
+        // add btn listener for confirm saving birthday
+        addBdayBinding.btnAddBdayAddContact.setOnClickListener { saveBday() }
+
+        // debug address for easy testing
+        if (BuildConfig.DEBUG) {
+            addBdayBinding.inputAddBdayContactName.setText("Kelvin")
+            addBdayBinding.inputAddBdayContactAddr.setText("742 evergreen terrace")
+        }
+    }
+
+    /**
+     * Populates the month birthday picker with array data onViewCreated
+     * Adds listener to change the date to the appropriate range based on month selected
+     */
+    private fun addDatePicker(context: Context) {
+        addBdayBinding.spinAddBdayMonth.adapter =
+            ArrayAdapter(context, android.R.layout.simple_spinner_item, Month.monthList)
+
+        addBdayBinding.spinAddBdayMonth.onItemSelectedListener = object: AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+                Timber.d("Nothing selected")
+            }
+
+            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+                changeBdayDateRange(Month.monthList[position])
+            }
+        }
+    }
+
+    /**
+     * Watches for add birthday saving results
+     */
+    private fun observeAddContact(contactAddedResult: AddBirthdayViewModel.ContactAddedResult?) {
```

Comment:
We can probably have `AddBirthdayViewModel.ContactAddedResult` be non-nullable here, since `addContactResult` isn't nullable in the view model class